mypy/subtypes.py

```diff
@@ -141,6 +141,8 @@ def visit_deleted_type(self, left: DeletedType) -> bool:
 
     def visit_instance(self, left: Instance) -> bool:
         if left.type.fallback_to_any:
+            if isinstance(self.right, NoneTyp):
```

Comment:
Might be worth adding a comment here explaining what is special about `None` here, and why we care, because this is a bit ad-hoc. (Though I suppose most things about `fallback_to_any` are a bit ad-hoc?)

Reply:
Makes sense, will add a comment.